moonraker/components/td1.py

```diff
@@ -0,0 +1,341 @@
+        self._watch_task = self._loop.create_task(self._watch_for_new_devices())
+
+    async def _start_all_serial_tasks(self) -> None:
+        devices = self._find_td1_devices()
+        if not devices:
+            self.logger.warning("No TD1 devices found.")
+            return
+        for port, serial_number in devices:
+            await self._start_serial_task(port, serial_number)
+            await asyncio.sleep(0.5)
+
+    async def _watch_for_new_devices(self) -> None:
+        self._known_serials = set(self.serial_tasks.keys())
+        while self._enabled:
+            await asyncio.sleep(5)  # Scan every 5 seconds
+
+            # Find currently connected devices
+            current_serials = set()
+            for port, serial_number in self._find_td1_devices():
+                current_serials.add(serial_number)
+                if serial_number not in self._known_serials:
+                    self.logger.info(f"Hot-plugged TD1 detected: {serial_number}")
+                    try:
+                        await self._start_serial_task(port, serial_number)
+                    except asyncio.CancelledError:
+                        raise
+                    except Exception:
+                        continue
+
+            # Detect disconnected devices
+            disconnected_serials = self._known_serials - current_serials
+            self.logger.debug(disconnected_serials, self._known_serials)
+            for serial_number in disconnected_serials:
+                if serial_number in self.serial_tasks:
+                    await self.close_device(self.serial_tasks[serial_number])
+                    del self.serial_tasks[serial_number]
+                self._known_serials.remove(serial_number)
+
+    def _find_td1_devices(self) -> list:
+        found = []
+        for port in serial.tools.list_ports.comports():
+            if port.vid == self._vid and port.pid == self._pid:
+                if port.serial_number:
+                    found.append((port.device, port.serial_number))
+        return found
+
+    async def _start_serial_task(self, port: str, serial_number: str) -> None:
+        try:
+            self.serial_tasks[serial_number] = TD1Protocol(self._config,
+                                                           serial_number,
+                                                           port,
+                                                           self.logger)
+            ret = self.serial_tasks[serial_number].initialize()
+            if ret is not None:
+                await ret
+
+            if not self.serial_tasks[serial_number].error_state:
+                self.logger.info(f"TD1 listening on {port} (Serial: {serial_number})")
+                self._known_serials.add(serial_number)
+            else:
+                self.logger.error(
+                    f"Failed to open port:{port} for TD1 serial {serial_number}")
+
+        except serial.serialutil.SerialException as e:
```

Comment:
I missed this before, however it appears that `serialutil` is not a valid attribute of `serial`.  This should be `serial.SerialException`.